front-end/src/screens/AddressLoginForm/index.tsx

```diff
@@ -0,0 +1,218 @@
+import getEncodedAddress from '../../util/getEncodedAddress';
+import cleanError from '../../util/cleanError';
+import { UserDetailsContext } from '../../context/UserDetailsContext';
+
+interface Props {
+	className?: string
+}
+
+const APPNAME = process.env.REACT_APP_APPNAME || 'polkassembly';
+
+const LoginForm = ({ className }:Props): JSX.Element => {
+	const [address, setAddress] = useState<string>('');
+	const [accounts, setAccounts] = useState<InjectedAccountWithMeta[]>([]);
+	const [extensionNotFound, setExtensionNotFound] = useState(false);
+	const [accountsNotFound, setAccountsNotFound] = useState(false);
+	const { history } = useRouter();
+	const { queueNotification } = useContext(NotificationContext);
+	const [addressLoginStartMutation] = useAddressLoginStartMutation();
+	const [addressLoginMutation, { loading, error }] = useAddressLoginMutation();
+	const currentUser = useContext(UserDetailsContext);
+
+	const getAccounts = async (): Promise<undefined> => {
+		const extensions = await web3Enable(APPNAME);
+
+		if (extensions.length === 0) {
+			setExtensionNotFound(true);
+			return;
+		} else {
+			setExtensionNotFound(false);
+		}
+
+		const accounts = await web3Accounts();
+
+		if (accounts.length === 0) {
+			setAccountsNotFound(true);
+			return;
+		} else {
+			setAccountsNotFound(false);
+		}
+
+		accounts.forEach((account) => {
+			account.address = getEncodedAddress(account.address) || account.address;
+		});
+
+		setAccounts(accounts);
+		if (accounts.length > 0) {
+			setAddress(accounts[0].address);
+		}
+
+		return;
+	};
+
+	const onAccountChange = (event: React.SyntheticEvent<HTMLElement, Event>, data: DropdownProps) => {
+		const addressValue = data.value as string;
+		setAddress(addressValue);
+	};
+
+	const navigateLogin = () => {
+		history.push('/login');
+	};
+
+	const handleLogin = async () => {
+		if (!accounts.length) {
+			return getAccounts();
+		}
+
+		try {
+			const injected = await web3FromSource(accounts[0].meta.source);
+			const signRaw = injected && injected.signer && injected.signer.signRaw;
+
+			if (!signRaw) {
+				return console.error('Signer not available');
+			}
+
+			const { data: startResult } = await addressLoginStartMutation({
+				variables: {
+					address
+				}
+			});
+
+			const signMessage = startResult?.addressLoginStart?.signMessage;
+
+			if (!signMessage) {
+				throw new Error('Challenge message not found');
+			}
+
+			const { signature } = await signRaw({
+				address,
+				data: stringToHex(signMessage),
+				type: 'bytes'
+			});
+
+			const { data: loginResult } = await addressLoginMutation({
+				variables: {
+					address,
+					signature
+				}
+			});
+
+			if (loginResult?.addressLogin?.token && loginResult?.addressLogin?.user) {
+				handleLoginUser({ token: loginResult.addressLogin.token, user: loginResult.addressLogin.user }, currentUser);
+				history.push('/');
+			} else {
+				throw new Error('Web3 Login failed');
+			}
+		} catch (error) {
```

Comment:
We should show the error in a message in the UI also, notifications are to quick to disapear I think.

Reply:
Forget about that, it's done, I havn't seen it.